### PR TITLE
Feature: Add most searched place in user profile page

### PR DESCRIPTION
**Pull request by timwangmusic**
## Description
Enrich user profile page by adding a favorite section. This PR adds the most searched place by a user.

## Solution
* Add a new data structure `PersonalFavorites` to the user view.
* Count and record the timestamp when a user searches a place.
* Frontend renders the user favorites besides saved travel plans.

## Testing
- [x] Integration testing on Heroku staging
- [x] Added new unit tests

## Checks
- [x] Have you removed commented code?
- [x] Have you used gofmt to format your code?

### CODE REVIEW

File: assets/js/user_profile.js [hunk 1]
```diff
@@ -67,4 +79,24 @@ async function getUserPlans() {
...
+            const favorites = data['searchHistory'];
+            if (Object.keys(favorites).length > 0) {
+                renderFavorites(favorites);
+            }
+        }
+    ).catch(
+        err => console.error(err)
+    );
+}
+
```
Comment (sananand007) [inline assets/js/user_profile.js:98]: wondering what is the difference if we do 
```
async function getUserFavorites() {
	const favoritesUrl = `/v1/users/${username}/favorites`;
	try {
		response = await axios.get(favoritesUrl)
		const data = response.data;
        const favorites = data['searchHistory'];
        if (Object.keys(favorites).length > 0) {
                renderFavorites(favorites);
        }
	}
	catch (error) {
		console.error(`Err: ${err}`)
	}
}
```

**Review by sananand007 — APPROVED** [review]
lgtm

File: assets/js/user_profile.js (hunk 1 above)
Comment (timwangmusic) [reply assets/js/user_profile.js:98]: `Axios` is implicitly doing the `try...catch...`.

**Review by rwangsc18 — COMMENTED** [review]
LGTM. 
It is difficult to optimize HTML/CSS layout for the favorite box without a signed-in user, because the staging environment doesn't allow creating a user. If okay, let's update the layout in a separate PR after we merge this PR to prod.

**Pull request merged**